src/libcore/num/mod.rs

```diff
@@ -100,6 +100,28 @@ assert_eq!(size_of::<Option<core::num::", stringify!($Ty), ">>(), size_of::<", s
                 }
             }
 
+            #[stable(feature = "from_nonzeroopt", since = "1.43.0")]
+            impl From<Option<$Ty>> for $Int {
+                doc_comment! {
+                    concat!(
+"Converts an `Option<", stringify!($Ty), ">` into an `", stringify!($Int), "`"),
+                    fn from(nonzero: Option<$Ty>) -> Self {
+                        nonzero.map($Ty::get).unwrap_or(0)
```

Comment:
As @timvermeulen said, this should be `nonzero.map_or(0, $Ty::get)`

Reply:
Ok I fixed it